kernel/fs/iso9660.inc

```diff
@@ -736,3 +736,1044 @@ cd_compare_name:
+
+        push    ebx
+        sub     esp, 4*3 ; locals value: primary_lba, encoding, select_sector
+        mov     dword[esp], 16-1 ; sector number
+        mov     eax, [esi + DISK.MediaInfo.LastSessionSector]
+        add     [esp], eax
+        mov     dword[esp + 4], 0 ; base encoding - ascii
+
+        add     ebx, CDBlockSize
+.new_descr:
+        inc     dword[esp]
+        ; read 16 sector, check header of descriptor
+        xor     edx, edx
+        mov     eax, [esp]
+        mov     ecx, 1
+        ; ebx - buffer
+        ; edx:eax - num sector
+        ; ebp - PARTITION
+        ; ecx - number sectors
+        call    fs_read64_sys
+        test    eax, eax
+        jnz     .err_disk_1
+
+        cmp     dword[ebx + 1], 'CD00'
+        jnz     .err_disk_1
+        cmp     byte[ebx + 5], '1'
+        jnz     .err_disk_1
+
+        cmp     byte[ebx], DESCRIPTOR_TYPE_TERMANATOR
+        jz      .end_scan
+
+        cmp     byte[ebx], DESCRIPTOR_TYPE_PRIMARY
+        jnz     @f
+
+        cmp     dword[esp + 4], 0
+        jnz     .new_descr
+
+        mov     dword[esp + 4], 0 ; set UCS-2 encoding
+        mov     eax, [esp]
+        mov     dword[esp + 8], eax ; set lba address
+
+        jmp     .new_descr
+@@:
+        cmp     byte[ebx], DESCRIPTOR_TYPE_SUPPLEMENTARY
+        jnz     .new_descr
+
+        mov     dword[esp + 4], 1 ; set UCS-2 encoding
+        mov     eax, [esp]
+        mov     dword[esp + 8], eax ; set lba address
+
+        jmp     .new_descr
+.end_scan:
+
+        ; get root dir lba, root dir size, size lba
+        xor     edx, edx
+        mov     eax, [esp + 8]
+        mov     ecx, 1
+        call    fs_read64_sys
+        test    eax, eax
+        jnz     .err_disk_1
+
+        ; alloc memory for ISO9660 struct
+        mov     eax, sizeof.ISO9660
+        call    malloc
+        test    eax, eax
+        jz      .err_disk_1
+        ; copy data on struct
+        add     esp, 4
+        pop     dword[eax + ISO9660.type_encoding]
+        mov     dword[eax + ISO9660.lba_size], CDBlockSize
+        pop     dword[eax + ISO9660.primary_descr]
+
+        mov     ecx, dword[ebx + ISO9660_PRIMARY_DESCRIPTOR.root_dir_record + ISO9660_DIRECTORY_RECORD.lba]
+        mov     [eax + ISO9660.root_dir_lba], ecx
+
+        mov     ecx, dword[ebx + ISO9660_PRIMARY_DESCRIPTOR.root_dir_record + ISO9660_DIRECTORY_RECORD.data_length]
+        mov     [eax + ISO9660.root_dir_len], ecx
+
+        push    edi esi
+        mov     edi, eax
+        mov     esi, ebp
+        mov     ecx, sizeof.PARTITION/4
+        rep movsd
+        pop     esi edi
+        mov     [eax + PARTITION.FSUserFunctions], iso9660_user_functions
+
+        pop     ebx
+        ret
+.err_disk_1:
+        add     esp, 4*3
+        pop     ebx
+.fail_disk_sector:
+        xor     eax, eax
+        ret
+
+; IN: eax - ptr PARTITION
+; OUT: -
+; SAVE: esi, edi
+;  Function free PARTITION struct and all object this structure
```

Comment:
It's not in English, just don't understand what was meant here, correct please you can use translate